### PR TITLE
Use bookdown 2 23

### PR DESCRIPTION
**Pull request by mitzimorris**
#### Submission Checklist

- [x] Builds locally 
- [x] Declare copyright holder and open-source license: see below

#### Summary

Updated `_output.yml`  files for all docs so that docs will build with bookdown 2.23.  Unfortunately, bookdown 2.23 introduced a breaking change for the yaml headers w/r/t search engine config.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by mitzimorris** [comment]
Improved `build.py` script and added more info to README.md.

In order to check for bookdown 2.23 or greater, using`subprocess.run` which requires Python 3.5 or higher.

**Comment by mitzimorris** [comment]
this is ready for review.

**Comment by WardBrian** [comment]
I get an error straight off:

```
(r-env) [brian@FlatTop bookdowntest on use-bookdown-2-23] $ python build.py 2 27
Traceback (most recent call last):
  File "/home/brian/Dev/integration/bookdowntest/build.py", line 125, in <module>
    main()
  File "/home/brian/Dev/integration/bookdowntest/build.py", line 77, in main
    check_bookdown_2_23()
  File "/home/brian/Dev/integration/bookdowntest/build.py", line 39, in check_bookdown_2_23
    check = subprocess.run(command, capture_output=True, text=True)
  File "/home/brian/miniconda3/envs/r-env/lib/python3.9/subprocess.py", line 505, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/home/brian/miniconda3/envs/r-env/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/brian/miniconda3/envs/r-env/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/bin/Rscript'
```
The reason seems to be the hard coding of the Rscript path:

```
(r-env) [brian@FlatTop bookdowntest on use-bookdown-2-23] $ which Rscript
/home/brian/miniconda3/envs/r-env/bin/Rscript
```

File: README.md [hunk 1]
```diff
@@ -1,41 +1,92 @@
...
+The Stan documentation uses the [Lucida fonts](https://www.pctex.com/Lucida_Fonts.html),
+which must be [installed manually](https://tex.stackexchange.com/questions/88423/manual-font-installation).
+
+
+## Scripts to build and maintain the docset
+
+**`build.py`**
+
+The program `build.py` convert the markdown files under `src` to html and pdf and populates the `docs` dir with the generated documentation.
+Requires Python 3.5 or higher, due to call to `subproces.run`.
```
Comment (WardBrian) [inline README.md:44]: While `subprocess.run` was added in 3.5, the `capture_output` kwarg wasn't until 3.7 -- I think that's probably fine, but we should correct the minimum version here.

File: build.py [hunk 2]
```diff
@@ -29,11 +29,18 @@ def pushd(new_dir):
...
 
 def shexec(command):
-    returncode = subprocess.call(command, shell=True)
+    returncode = subprocess.call(command, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
     if returncode != 0:
         print('Command {} failed'.format(command))
         raise Exception(returncode)
 
+def check_bookdown_2_23():
+    command = ["/usr/local/bin/Rscript", "-e", "utils::packageVersion(\"bookdown\") > 0.22"]
```
Comment (WardBrian) [inline build.py:38]:
```suggestion
    command = ["Rscript", "-e", "utils::packageVersion(\"bookdown\") > 0.22"]
```

**Review by WardBrian — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mitzimorris) [reply README.md:44]: also checking in script itself.

File: build.py (hunk 2 above)
Comment (mitzimorris) [reply build.py:38]: doh!  done!

**Review by WardBrian — APPROVED** [review]

**Pull request merged**